File: backend/spring-app/spring-app/src/main/java/com/avdo/spring/app/entity/OrderItem.java
package com.avdo.spring.app.entity;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;

@Entity
@Table(name = "order_item")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class OrderItem {

    @Id
    @Column(name = "order_item_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "price")
    private double price;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private ProductEntity product;

    public OrderItem() {
    }

    public Long getId() {
        return this.id;
    }

    public Order getOrder() {
        return this.order;
    }

    public int getQuantity() {
        return this.quantity;
    }

    public double getPrice() {
        return this.price;
    }

    public ProductEntity getProduct() {
        return this.product;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public void setProduct(ProductEntity product) {
        this.product = product;
    }

    public String toString() {
        return "OrderItem(id=" + this.getId() +
                ", orderId=" + this.getOrder().getId() +
                ", quantity=" + this.getQuantity() +
                ", price=" + this.getPrice() +
                ", productId=" + this.getProduct().getId() + ")";
    }
}

File: frontend/react-project/src/models/OrderItem.ts
import Product from "./Product";

export default class OrderItem {

    private id: number;
    private orderId: number;
    private quantity: number;
    private price: number;
    private product: Product;

    public constructor(id: number, orderId: number, quantity: number, price: number, product: Product) {
        this.id = id;
        this.orderId = orderId;
        this.quantity = quantity;
        this.price = price;
        this.product = product;
    }

    public getId(): number {
        return this.id;
    }

    public setId(value: number) {
        this.id = value;
    }

    public getOrderId(): number {
        return this.orderId;
    }

    public setOrderId(value: number) {
        this.orderId = value;
    }

    public getQuantity(): number {
        return this.quantity;
    }

    public setQuantity(value: number) {
        this.quantity = value;
    }

    public getPrice(): number {
        return this.price;
    }

    public setPrice(value: number) {
        this.price = value;
    }

    public getProduct(): Product {
        return this.product;
    }
    public setProduct(value: Product) {
        this.product = value;
    }
}